src/utils/index.js

```diff
@@ -0,0 +1,106 @@
+'use strict'
+
+const async = require('async')
+const fs = require('fs')
+const hat = require('hat')
+const os = require('os')
+const path = require('path')
+const exec = require('../exec')
+const safeParse = require('safe-json-parse/callback')
+const createRepo = require('./create-repo-nodejs')
+
+const join = path.join
+
+const isWindows = os.platform() === 'win32'
+
+exports.createRepo = createRepo
+
+// taken from https://github.com/hughsk/flat
+exports.flatten = (target) => {
+  let output = {}
+  const step = (object, prev) => {
+    object = object || {}
+    Object.keys(object).forEach(function (key) {
+      let value = object[key]
+      let isarray = Array.isArray(value)
+      let type = Object.prototype.toString.call(value)
+      let isbuffer = Buffer.isBuffer(value)
+      let isobject = (
+        type === '[object Object]' ||
+        type === '[object Array]'
+      )
+
+      let newKey = prev
+        ? prev + '.' + key
+        : key
+
+      if (!isarray && !isbuffer && isobject && Object.keys(value).length) {
+        return step(value, newKey)
+      }
+
+      output[newKey] = value
+    })
+  }
```

Comment:
all of the `let` could be `const` here in `exports.flatten`, if I read this correctly, but it shouldn't really matter.

Reply:
https://github.com/ipfs/js-ipfsd-ctl/blob/a90c19ae74e560c82cd91d6b4fbdb7fd863062c7/src/utils/index.js#L20-L35

Reply:
good catch!

Reply:
this is fixed in the latest version